### PR TITLE
[NFC] implement clippy changes in llvm_interpreter

### PR DESCRIPTION
**Pull request by bollu**

### CODE REVIEW

File: src/llvm_interpreter.rs [hunk 1]
```diff
@@ -702,6 +702,8 @@ struct LLVMBoxedStructRepr {
 }
 
 impl LLVMJIT {
+    //FIXME: not sure why this is seen as a "useless" transmute, I need to cast it to a void*
+    #[cfg_attr(feature = "cargo-clippy", allow(useless_transmute))]
```
Comment (polybuildr) [inline src/llvm_interpreter.rs]: What clippy suggests is: `libc::malloc as usize as *mut libc::c_void` which seems weird, but I guess makes sense.

Also, you can actually use this attribute on only the actual function call by placing it on the line before the `mem::transmute`.

**Review by polybuildr — COMMENTED** [review]
Looks good. Left one comment inline - let's discuss the FIXME there and then merge.

File: src/llvm_interpreter.rs (hunk 1 above)
Comment (bollu) [reply src/llvm_interpreter.rs]: that's strange, but I don't mind, I guess.

Comment (polybuildr) [reply src/llvm_interpreter.rs]: Your choice - I'm fine with this either way.

**Comment by polybuildr** [comment]
Looks good. Will merge after tests complete, just to be safe.

**Pull request merged**